src/nomad_simulations/schema_packages/properties/energies.py

```diff
@@ -20,33 +20,65 @@
 
 import numpy as np
 
-from nomad.metainfo import Quantity
+from nomad.metainfo import Quantity, Section, Context, SubSection
 
 if TYPE_CHECKING:
     from nomad.metainfo import Section, Context
     from nomad.datamodel.datamodel import EntryArchive
     from structlog.stdlib import BoundLogger
 
-from nomad_simulations.schema_packages.physical_property import PhysicalProperty
+from nomad_simulations.schema_packages.physical_property import (
+    PhysicalProperty,
+    PropertyContribution,
+)
 
+##################
+# Abstract classes
+##################
 
-class FermiLevel(PhysicalProperty):
+
+class BaseEnergy(PhysicalProperty):
     """
-    Energy required to add or extract a charge from a material at zero temperature. It can be also defined as the chemical potential at zero temperature.
+    Abstract physical property section describing some energy of a (sub)system.
     """
 
-    # ! implement `iri` and `rank` as part of `m_def = Section()`
-
-    iri = 'http://fairmat-nfdi.eu/taxonomy/FermiLevel'
-
     value = Quantity(
         type=np.float64,
         unit='joule',
         description="""
-        The value of the Fermi level.
+        The value of the energy.
```

Comment:
For the value description suggestion, I think we need to decide on some convention for this. These descriptions are essentially redundant because the class description is also describing the value.

The 2nd point is for @JosePizarro3

Reply:
Indeed, the description of the `value` is the same as the PhysicalProperty. We can leave it empty if needed.

> Btw, here you could also specify the sign convention. Or maybe you consider the vacuum as a reference, and all energies should be positive?

Actually, I think the Fermi level (whatever is parsed from the outputs) should be the reference in electronic properties. But I'd might be wrong, and some code might be doing something different that might affect this idea.

Reply:
> Indeed, the description of the value is the same as the PhysicalProperty. We can leave it empty if needed.

Don't you need to overwriet the attributes from `self.value` in `__init__()` then? Atm, you are overwriting the whole quantity.

Reply:
I opted to remove all my value descriptions for now.

I am not going to do anything with the FermiLevel here, so if something needs to be done you should open an issue